csm/cli/support_bundle/support_bundle.py

```diff
@@ -53,56 +53,30 @@ def import_provisioner_plugin():
+    async def fetch_active_nodes_hosts():
         """
-        This method fetch hostname for all nodes from Salt DB
+        This Method is for reading hostnames, node_list information.
         :return: hostnames : List of Hostname :type: List
         :return: node_list : : List of Node Name :type: List
         """
-        process = SimpleProcess(
-            "salt-call pillar.get cluster:node_list --out=json")
-        _out, _err, _rc = process.run()
-        if _rc != 0:
-            Log.warn(f"Salt command failed : {_err}")
-            return None, None
-        output = JsonMessage(_out.strip()).load()
-        nodes = output.get("local", [])
-        hostnames = []
-        for each_node in nodes:
-            process = SimpleProcess(
-                f"salt-call pillar.get cluster:{each_node}:hostname --out=json")
-            _out, _err, _rc = process.run()
-            if _rc != 0:
-                Log.warn(f"Salt command failed : {_err}")
-                return None, None
-            output = JsonMessage(_out.strip()).load()
-            hostnames.append(output.get("local", ""))
-        return hostnames, nodes
-
-    @staticmethod
-    async def fetch_host_from_cluster():
-        """
-        This Method is Backup method for Reading Cluster.sls if Salt Read Fails.
-        :return: hostnames : List of Hostname :type: List
-        :return: node_list : : List of Node Name :type: List
-        """
-        Log.info("Falling back to reading cluster information from cluster.sls.")
-        cluster_file_path = Conf.get(const.CSM_GLOBAL_INDEX,
-                                     "SUPPORT_BUNDLE>cluster_file_path")
-        if not cluster_file_path or not os.path.exists(cluster_file_path):
-            repsonse_msg = {"message": (f"{cluster_file_path} not Found. \n"
-                                        f"Please check if cluster info file is correctly configured.")}
-            return Response(rc=errno.ENOENT, output=repsonse_msg), None
-        cluster_info = Yaml(cluster_file_path).load().get("cluster", {})
-        active_nodes = cluster_info.get("node_list", [])
-        if not active_nodes:
+        Log.info("reading hostnames, node_list information")
+        node_hostname_map = dict()
+        mapping_dict = dict()
+        mapping_dict = Conf.get(const.CSM_GLOBAL_INDEX, f"{const.MAINTENANCE}")
+        if not mapping_dict:
+            response_msg = {
+                "message": "No MAINTENANCE dictionary found. csm.conf file may not be valid"}
+            return Response(output=response_msg,
+                            rc=CSM_ERR_INVALID_VALUE), None
+        node_hostname_map = mapping_dict.copy()
+        node_hostname_map.pop(const.SHUTDOWN_CRON_TIME)
+        if not node_hostname_map:
             response_msg = {
-                "message": "No active nodes found. Cluster file may not be valid"}
+                "message": "No node_hostname_map found. csm.conf file may not be valid"}
```

Comment:
No node_hostname_map found. csm.conf file may not be valid ---> Node list and hostname not found.

Reply:
done @pawankumarsrivastava